### PR TITLE
Add more hw functions and generic system for deviceAccess

### PR DESCRIPTION
**Pull request by gre**

### CODE REVIEW

File: src/hw/installOsuFirmware.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import ManagerAPI from "../api/Manager";
+
+export default (
+  transport: Transport<*>,
+  targetId: string | number,
+  firmware: OsuFirmware
+): Observable<*> => {
+  const params = {
+    targetId,
+    ...firmware,
```
Comment (valpinkman) [inline src/hw/installOsuFirmware.js]: this is a bit nasty as we also send all the applications array as URL argument, which can be quite long... I guess it's safer/cleaner to just explode the keys we need (targetId, firmware, firmwareKey and perso)

File: src/hw/firmwareUpdate-prepare.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+// @flow
+import { Observable, from, of, empty, concat } from "rxjs";
+import { mergeMap, delay, filter, map, throttleTime } from "rxjs/operators";
+
+import getDeviceInfo from "../hw/getDeviceInfo";
+import installOsuFirmware from "../hw/installOsuFirmware";
+import { withDevice } from "../hw/deviceAccess";
+import type { OsuFirmware } from "../types/manager";
+
+const wait3s = of({ type: "wait" }).pipe(delay(3000));
```
[the remark this answers is not in the document]
Comment (gre) [reply src/hw/firmwareUpdate-prepare.js]: would probably be overkill for the localized usage we have here

[the remark this answers is not in the document]
Comment (valpinkman) [reply src/hw/firmwareUpdate-prepare.js]: fine with me :)

File: src/hw/installOsuFirmware.js (hunk 1 above)
Comment (valpinkman) [reply src/hw/installOsuFirmware.js]: I'll take it

Comment (valpinkman) [reply src/hw/installOsuFirmware.js]: seems like it's done already

**Pull request merged**